torch/csrc/jit/custom_graph_executor_impl.cpp

```diff
@@ -0,0 +1,57 @@
+#include <torch/csrc/jit/custom_graph_executor_impl.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+struct GraphExecutorImplState {
+  Symbol name = kDefaultExecutor;
+  bool impl_cached = false;
+  GraphExecutorImplCreator impl_fn;
+};
+thread_local GraphExecutorImplState executor_state;
```

Comment:
Hmm I'm not sure if we really want this to be thread local. cc @zdevito

Reply:
Zach actually insisted making a profiling flag `thread_local`. Although, this isn't exactly the same use pattern 😄